src/radix_connect/mobile/client.rs

```diff
@@ -1,49 +1,174 @@
-//     // RadixConnectMobile should require a NetworkAntenna and a SecureStorage from the Wallet.
-//     // The internal components, such as RadixConnectRelayService will be created by the RadixConnectMobile.
-//     #[uniffi::constructor]
-//     pub fn new(
-//         _network_antenna: Arc<dyn NetworkAntenna>,
-//         _secure_storage: Arc<dyn SecureStorage>,
-//     ) -> Self {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn handle_linking_request(
-//         &self,
-//         _request: RadixConnectMobileLinkRequest,
-//     ) -> Result<Url> {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn handle_dapp_interaction_request(
-//         &self,
-//         _request: RadixConnectMobileDappRequest,
-//     ) -> Result<DappToWalletInteraction> {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn send_dapp_interaction_response(
-//         &self,
-//         _response: WalletToDappInteractionResponse,
-//     ) -> Result<Url> {
-//         todo!()
-//     }
-// }
+#[derive(uniffi::Object)]
+pub struct RadixConnectMobile {
+    /// The Radix Connect Relay service to be used to communicate with dApps.
+    relay_service: RelayService,
+    /// The secure storage to be used to store session data.
+    secure_storage: WalletClientStorage,
+    /// The new sessions that have been created and are waiting to be validated on dApp side.
+    /// Once the session is validated, it will be moved to the secure storage.
+    /// Validation consists in verifying the origin of the session.
+    new_sessions: RwLock<HashMap<SessionID, Session>>,
+}
+
+// Provisional API
+#[uniffi::export]
+impl RadixConnectMobile {
+    // RadixConnectMobile should require a NetworkAntenna and a SecureStorage from the Wallet.
+    // The internal components, such as RadixConnectRelayService will be created by the RadixConnectMobile.
+    #[uniffi::constructor]
+    pub fn new(
+        network_antenna: Arc<dyn NetworkAntenna>,
+        secure_storage: Arc<dyn SecureStorage>,
+    ) -> Self {
+        Self {
+            relay_service: RelayService::new_with_network_antenna(
+                network_antenna,
+            ),
+            secure_storage: WalletClientStorage::new(secure_storage),
+            new_sessions: RwLock::new(HashMap::new()),
+        }
+    }
+
+    #[uniffi::method]
+    pub async fn handle_linking_request(
+        &self,
+        request: RadixConnectMobileLinkRequest,
+    ) -> Result<Url> {
+        // 1. Get the handshake request from the relay service
+        let handshake_request = self
+            .relay_service
+            .get_session_handshake_request(request.session_id)
+            .await?;
+
+        // 2. Generate a new Diffie-Hellman key pair
+        let wallet_private_key = DiffieHellmanPrivateKey::generate()?;
+
+        // 3. Compute the shared secret
+        // random salt
+        let salt = hex_decode("000102030405060708090a0b0c0d0e0f").unwrap();
```

Comment:
@CyonAlexRDX What values are the most appropriate and how should we store this? I am not sure these should be in plain sight, but maybe it is ok.

Reply:
The `salt` and the `info` must be known by both parties (wallet and web-counter-part) or something both can derive/calculate from shared knowledge.

Should this method, `handle_linking_request` save the `wallet_private_key` into `secure_storage`? or is it ephemeral? if it should be saved (that call is missing) then we should try to find a suitable salt/info. 

If it is ephemeral, salt and info is not as important.

So talk with @xstelea if there is any shared knowledge both parties have? Origin? Url? Some headers? Other metadata? Some stable onchain data?